frame/header-mmr/src/lib.rs

```diff
@@ -195,24 +253,24 @@ pub mod pallet {
 		}
 	}
 	impl<T: Config> MMRStore<T::Hash> for ModuleMMRStore<T> {
-		fn get_elem(&self, pos: u64) -> MMRResult<Option<T::Hash>> {
+		fn get_elem(&self, pos: NodeIndex) -> MMRResult<Option<T::Hash>> {
 			Ok(<Pallet<T>>::mmr_node_list(pos))
```

Comment:
For runtime storage, need to use peaks storages for get_elem, and both peaks and offchain index for append_elems, also carefully pruning the peaks then.

For offchain storage, for rpc read, need to use offchain index for get_elem.